pkg/log/zap/zap_test.go

```diff
@@ -296,3 +299,170 @@ var _ = Describe("Zap logger setup", func() {
+
+			outRaw := logOut.Bytes()
+
+			Expect(string(outRaw)).Should(ContainSubstring(info))
+			Expect(string(outRaw)).Should(ContainSubstring(debug1))
+
+		})
+
+		It("Should set error zap-log-level flags.", func() {
+			args := []string{"--zap-log-level=error"}
+			fromFlags.BindFlags(&fs)
+			err := fs.Parse(args)
+			Expect(err).ToNot(HaveOccurred())
+
+			logOut := new(bytes.Buffer)
+
+			logger := New(UseFlagOptions(&fromFlags), WriteTo(logOut))
+			logger.V(0).Info(info)
+			logger.V(1).Info(debug1)
+
+			outRaw := logOut.Bytes()
+
+			Expect(outRaw).To(BeEmpty())
+		})
+
+	})
+
+	Context("with  zap-log-level  with increased verbosity.", func() {
+		It("Should set integer debug level 1 for zap-log-level flag.", func() {
+			args := []string{"--zap-log-level=1"}
+			fromFlags.BindFlags(&fs)
+			err := fs.Parse(args)
+			Expect(err).ToNot(HaveOccurred())
+			logOut := new(bytes.Buffer)
+
+			logger := New(UseFlagOptions(&fromFlags), WriteTo(logOut))
+			logger.V(0).Info(info)
+			logger.V(1).Info(debug1)
+
+			outRaw := logOut.Bytes()
+
+			Expect(string(outRaw)).Should(ContainSubstring(info))
+			Expect(string(outRaw)).Should(ContainSubstring(debug1))
+		})
+
+		It("Should set integer debug level 1 for zap-log-level flag.", func() {
+			args := []string{"--zap-log-level=1", "--zap-devel=true"}
+			fromFlags.BindFlags(&fs)
+			err := fs.Parse(args)
+			Expect(err).ToNot(HaveOccurred())
+			logOut := new(bytes.Buffer)
+
+			logger := New(UseFlagOptions(&fromFlags), WriteTo(logOut))
+			logger.V(0).Info(info)
+			logger.V(1).Info(debug1)
+
+			outRaw := logOut.Bytes()
+
+			Expect(string(outRaw)).Should(ContainSubstring(info))
+			Expect(string(outRaw)).Should(ContainSubstring(debug1))
+		})
+
+		It("Should set integer debug level 2 for zap-log-level flag.", func() {
+			args := []string{"--zap-log-level=2", "--zap-devel=true"}
+			fromFlags.BindFlags(&fs)
+			err := fs.Parse(args)
+			Expect(err).ToNot(HaveOccurred())
+			logOut := new(bytes.Buffer)
+
+			logger := New(UseFlagOptions(&fromFlags), WriteTo(logOut))
+			logger.V(0).Info(info)
+			logger.V(1).Info(debug1)
+			logger.V(2).Info(debug2)
+
+			outRaw := logOut.Bytes()
+
+			Expect(string(outRaw)).Should(ContainSubstring(info))
+			Expect(string(outRaw)).Should(ContainSubstring(debug1))
+			Expect(string(outRaw)).Should(ContainSubstring(debug2))
+
+		})
+		It("Should set integer debug level 3 for zap-log-level flag.", func() {
+			args := []string{"--zap-log-level=3", "--zap-devel=false"}
+			fromFlags.BindFlags(&fs)
+			err := fs.Parse(args)
+			Expect(err).ToNot(HaveOccurred())
+			logOut := new(bytes.Buffer)
+
+			logger := New(UseFlagOptions(&fromFlags), WriteTo(logOut))
+			logger.V(0).Info(info)
+			logger.V(1).Info(debug1)
+			logger.V(3).Info(debug3)
+
+			outRaw := logOut.Bytes()
+
+			Expect(string(outRaw)).Should(ContainSubstring(info))
+			Expect(string(outRaw)).Should(ContainSubstring(debug1))
+			Expect(string(outRaw)).Should(ContainSubstring(debug3))
```

Comment:
Shows that the test is duplicated and this one is testing the previous scenarios as well. in this way, could we not keep this one only and then add tests to ensure that the logs do not appear when the level is not set?

Reply:
Previous test case checks for `args := []string{"--zap-log-level=2", "--zap-devel=true"}`, where as above test case checks for `args := []string{"--zap-log-level=3", "--zap-devel=false"}`. They are different test cases.

Reply:
We should make clear the test cases. E.g `should not output errors logs when the zap level informed in info`, should output debug logs when the zaplevel informed is info and the development mode is enable `. 

Also, what means level 3 if we have just error and info? (0,1)? 
What should be output when the debug mode `--zap-devel=true` is used or not in each case?

Reply:
We do have `debug` for `--zap-log-level` , please see below:
```var levelStrings = map[string]zapcore.Level{
	"debug": zap.DebugLevel,
	"info":  zap.InfoLevel,
	"error": zap.ErrorLevel,
}
```